docs/ml-clustering.md

```diff
@@ -265,3 +265,44 @@ Refer to the [R API docs](api/R/spark.gaussianMixture.html) for more details.
 </div>
 
 </div>
+
+## Power Iteration Clustering (PIC)
+
+Power Iteration Clustering (PIC) is  a scalable graph clustering algorithm
```

Comment:
could you open a separate PR with just this file (minus R) and FPGrowthExample.scala on branch-2.4?

Reply:
sure

Reply:
Pardon, I'm catching up -- why just commit this doc to 2.4 and not master?

Reply:
The doc change will be in both 2.4 and master, but the R related code will be in master only. I think that's why @felixcheung asked me to open a separate PR to merge in the doc change for 2.4.

Reply:
OK sounds good. Let's merge this one first just as a matter of process.